src/generic/DirectSum.jl

```diff
@@ -220,6 +224,11 @@ function direct_sum(m::Vector{<:AbstractAlgebra.FPModule{T}}) where T <: RingEle
    return M, inj, pro
 end
 
+function direct_sum(M::DirectSumModule{T}, N::DirectSumModule{T}, mp::Vector{ModuleHomomorphism{T}}) where T
```

Comment:
this function should have `hom` in its name. In Oscar, we call the similar function that lifts homs to a tensor product `hom_tensor`. For direct sums, there is the equivalent for Lie algebra modules as `hom_direct_sum`, but we can adapt the name of that in Oscar to match this one here. But IMO it should start with `hom_*`